official/transformer/transformer_main.py

```diff
@@ -388,20 +400,35 @@ def _check_train_limits(flag_dict):
     return flag_dict["train_epochs"] is None or flag_dict["train_steps"] is None
 
   @flags.multi_flags_validator(
-      ["data_dir", "bleu_source", "bleu_ref", "vocab_file"],
-      message="--bleu_source, --bleu_ref, and/or --vocab_file don't exist. "
+      ["bleu_source", "bleu_ref"],
+      message="Files specified by --bleu_source and/or --bleu_ref don't exist. "
               "Please ensure that the file paths are correct.")
   def _check_bleu_files(flags_dict):
     """Validate files when bleu_source and bleu_ref are defined."""
     if flags_dict["bleu_source"] is None or flags_dict["bleu_ref"] is None:
       return True
-    # Ensure that bleu_source, bleu_ref, and vocab files exist.
-    vocab_file_path = os.path.join(
-        flags_dict["data_dir"], flags_dict["vocab_file"])
     return all([
         tf.gfile.Exists(flags_dict["bleu_source"]),
-        tf.gfile.Exists(flags_dict["bleu_ref"]),
-        tf.gfile.Exists(vocab_file_path)])
+        tf.gfile.Exists(flags_dict["bleu_ref"])])
+
+  @flags.multi_flags_validator(
+      ["data_dir", "model_dir", "vocab_file"],
+      message="Please ensure that the --vocab_file exists in --data_dir. If it "
+              "exists, please make sure the contents match that of the vocab "
+              "file in --model_dir.")
```

Comment:
I also think the error message should provide action items to user to unblock them or fix the issue. In this case, probably a command to remove the vocab file in model is more appropriate.